marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/Database/PostgreSQL.hs

```diff
@@ -555,6 +559,48 @@ queryTxOuts slotNo txId = HT.statement (Api.unTxId txId, fromIntegral slotNo) $
               }
           }
 
+getUTxOs :: GetUTxOs Transaction
+getUTxOs = GetUTxOs \addresses -> do
+    let
+      addresses' = V.fromList $ fmap Api.unAddress. Set.toList $ addresses
+    HT.statement addresses' $
+      [foldStatement|
+        SELECT txOut.txId :: bytea
+             , txOut.txIx :: smallint
+             , txOut.address :: bytea
+             , txOut.lovelace :: bigint
+             , txOut.datumHash :: bytea?
+             , txOut.datumBytes :: bytea?
+             , asset.policyId :: bytea?
+             , asset.name :: bytea?
+             , assetOut.quantity :: bigint?
+          FROM chain.txOut         AS txOut
+          LEFT JOIN chain.assetOut AS assetOut ON assetOut.txOutId = txOut.txId AND assetOut.txOutIx = txOut.txIx
+          LEFT JOIN chain.asset    AS asset    ON asset.id = assetOut.assetId
+         WHERE txOut.address = ANY($1 :: bytea[])
+         ORDER BY txIx
+      |] (Fold foldRow mempty id)
+    where
+      -- foldRow :: IntMap Api.TransactionOutput -> ReadTxOutRow -> IntMap Api.TransactionOutput
```

Comment:
Why comment out the type signature?

Reply:
`ReadTxOut` is a bit different in this case I include `TxId` - I'm going to probably drop the signature all together.